libclc/ptx-nvidiacl/libspirv/math/fabs.cl

```diff
@@ -18,12 +18,16 @@
 
 // Requires at least sm_80
 _CLC_DEF _CLC_OVERLOAD ushort __clc_fabs(ushort x) {
-    return __nvvm_abs_bf16(x);
+    ushort res;
+    __asm__ volatile("abs.bf16 %0,%1;" : "=h"(res) : "h"(x));
+    return res;
 }
 _CLC_UNARY_VECTORIZE(_CLC_OVERLOAD _CLC_DEF, ushort, __clc_fabs, ushort)
 
 // Requires at least sm_80
 _CLC_DEF _CLC_OVERLOAD uint __clc_fabs(uint x) {
-    return __nvvm_abs_bf16x2(x);
+    uint res;
+    __asm__ volatile("abs.bf16x2 %0,%1;" : "=r"(res) : "r"(x));
```

Comment:
Are the `volatile` asm qualifiers necessary? Given that these assembly blocks all have output constraints, I don't think the volatile is necessary and it prevents the optimizer from doing potentially useful reordering

Reply:
From my understanding it is necessary, but perhaps I am being overly cautious? See https://docs.nvidia.com/cuda/inline-ptx-assembly/index.html#incorrect-optimization for more details.

Reply:
Yeah I think in this case it is safe because there are no side-effects. I can remove volatile everywhere.

Reply:
Yeah. I think that doc is referring to the fact that the order of `mov`ing from `%clock` is sensitive as `%clock` is itself a volatile register, but in this case I don't think we should have volatile.

Reply:
OK I'm removing all `volatile`